examples/with-sitemap/scripts/generate-sitemap.js

```diff
@@ -0,0 +1,35 @@
+const fs = require('fs')
+
+const globby = require('globby')
+
+const WEBSITE_URL = 'http://localhost:3000'
+
+;(async () => {
+  // Ignore Next.js specific files (e.g., _app.js) and API routes.
+  const pages = await globby([
+    'pages/**/*{.js,.mdx}',
+    '!pages/_*.js',
+    '!pages/api',
+  ])
+  const sitemap = `
+  <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+            ${pages
+              .map((page) => {
+                const path = page
+                  .replace('pages', '')
+                  .replace('.js', '')
+                  .replace('.mdx', '')
+                const route = path === '/index' ? '' : path
+
+                return `
+                        <url>
+                            <loc>${`${WEBSITE_URL}${route}`}</loc>
+                        </url>
+                    `
+              })
+              .join('')}
+              </urlset>
+    `
+
+  fs.writeFileSync('public/sitemap.xml', sitemap)
+})()
```

Comment:
Updated to produce a file with better tabs, to have a simpler syntax and to use an environment variable for the website url

**Important: Add a `.env` file with the following content after accepting this change:**

```bash
# Used to add the domain to sitemap.xml, replace it with a real domain in production
WEBSITE_URL=http://localhost:3000
```

```suggestion
const fs = require('fs')
const globby = require('globby')

function addPage(page) {
  const path = page.replace('pages', '').replace('.js', '').replace('.mdx', '')
  const route = path === '/index' ? '' : path

  return `  <url>
    <loc>${`${process.env.WEBSITE_URL}${route}`}</loc>
    <changefreq>hourly</changefreq>
  </url>`
}

async function generateSitemap() {
  // Ignore Next.js specific files (e.g., _app.js) and API routes.
  const pages = await globby([
    'pages/**/*{.js,.mdx}',
    '!pages/_*.js',
    '!pages/api',
  ])
  const sitemap = `<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
${pages.map(addPage).join('\n')}
</urlset>`

  fs.writeFileSync('public/sitemap.xml', sitemap)
}

generateSitemap()

```